### PR TITLE
Fixed Numerous Issues 1)added MAXHP 2)Added HUD Msgs (Item Discovery,…

### PR DESCRIPTION
**Pull request by mberlanga**
… Item Pickup, and Wall Boundary Warning) 3) Added 'l' linger button 4) Added Diagonal Movement with 'q','e','z',and 'c' 5)Fixed Level Coloring Bug 6) Prevented Player from using a turn to run into a wall 7)Fixed Major Bug with monster dissapearing off-screen when near walls or other monsters

#87 
#86 
#78 
#71 
#70 
#65 
#69

### CODE REVIEW

File: src/edu/ucsb/cs56/projects/games/roguelike/LogicEngine.java [hunk 1]
```diff
@@ -65,6 +66,37 @@ public LogicEngine(int width, int height){
 		level = 1;
 		elixirStepCounter = 0;
 	}
+
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/LogicEngine.java]: What's with all the whitespace!?

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java [hunk 2]
```diff
@@ -363,26 +533,64 @@ public void keyPressed(KeyEvent key){
...
+		if (x != origX && y!=origY){
+		//x = (x > origX) ? x-1 : x+1;
+		x = origX;
+		y = origY;
+	    }
+	    else if (y != origY)
+		//y = (y > origY) ? y-1 : y+1;
+		y = origY;
+	    else if (x!= origX) x = origX;
+	}
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: What's this bracket associated with?

File: src/edu/ucsb/cs56/projects/games/roguelike/Player.java [hunk 3]
```diff
@@ -17,12 +17,13 @@
     private int score;
     private String typeOfPiece;
     private char icon;
+    final static public int MAXHP = 100;// this is NEW
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/Player.java]: Let's remove the comment.  It makes sense as a PR comment, but will be confusing it left in the code.

File: src/edu/ucsb/cs56/projects/games/roguelike/Player.java [hunk 4]
```diff
@@ -17,12 +17,13 @@
...
 
     /**
      * creates a player with 100 hitPoints, 20 attack, and 1 speed
      */
     public Player(){
-	this.hitPoints = 100;
+	this.hitPoints = MAXHP; //this is NEW
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/Player.java]: Ditto.

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java [hunk 5]
```diff
@@ -62,12 +80,28 @@ public RogueController(){
...
+
+
+    
     
     /**
      * Handles movement of player by checking if it can move there first through the logic engine
      * if it can move, invoke the canvas to animate it
      * if it can't, it checks if its because of out of bounds or a monster
-     * if its a monster the player will attack it and its its dead the canvas will animate the removal of the monster
+     * if its a monster the player will attack it and if it's dead the canvas will animate the removal of the monster
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: 👍

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java [hunk 6]
```diff
@@ -148,6 +200,10 @@ public void moveMonster(){
 		    } else if (logicHandler.movable(xPos, yPos)) {
 			logicHandler.move(xPos, yPos,xOrig,yOrig);
 			canvas.moveMonster(xPos, yPos,logicHandler.getObject(xPos,yPos));
+		    }else if ((!logicHandler.movable(xPos,yPos))){
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: Shouldn't this just be an `else`?

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java [hunk 7]
```diff
@@ -148,6 +200,10 @@ public void moveMonster(){
...
+			logicHandler.move(xOrig, yOrig,xOrig,yOrig);
+			canvas.moveMonster(xOrig, yOrig,logicHandler.getObject(xOrig,yOrig));
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: And what's the point of "moving" a monster to the same slot?

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java [hunk 8]
```diff
@@ -363,26 +533,64 @@ public void keyPressed(KeyEvent key){
 	origX = x; 
 	origY = y;
 	switch (key.getKeyChar()){
-	case 'w'	    :	 this.y = this.y - logicHandler.getPlayer().getSpeed(); break;
-	case 'a'	    :  	 this.x = this.x - logicHandler.getPlayer().getSpeed(); break;
-	case 'd'	    :	 this.x = this.x + logicHandler.getPlayer().getSpeed(); break;
-	case 's'		:	 this.y = this.y + logicHandler.getPlayer().getSpeed(); break;
+	case 'w'	    :
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: Can you fix the weird spacing here?  It's usually `case 'w':` or `default:`.

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java [hunk 9]
```diff
@@ -363,26 +533,64 @@ public void keyPressed(KeyEvent key){
...
+		this.y = this.y - logicHandler.getPlayer().getSpeed(); break;
+	case 'q'	    :
+		this.y = this.y - logicHandler.getPlayer().getSpeed();
+		this.x = this.x - logicHandler.getPlayer().getSpeed(); break;
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: Now that these are multi-line, might as well throw the `break` on its own line.  It's easy to miss it this way.

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java [hunk 10]
```diff
@@ -363,26 +533,64 @@ public void keyPressed(KeyEvent key){
...
-	moveHero();
-	moveMonster();
+	//Tell player they cannot run into a wall, else continue with game progress
+	if(logicHandler.getObject(x,y) instanceof Wall){
+	    canvas.write("     Ouch a Wall  ", 61,23,RoguePanel.yellow,RoguePanel.red);
+
+	    //restores player (x,y) player position if a wall is hit
+	    while (!logicHandler.isGround(x, y)) {
+		if (x != origX && y!=origY){
+		//x = (x > origX) ? x-1 : x+1;
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: Strip out your dead code.

File: src/edu/ucsb/cs56/projects/games/roguelike/RoguePanel.java [hunk 11]
```diff
@@ -545,7 +545,7 @@ public RoguePanel write(char character) {
      * @param background the background color or null to use the default
      * @return this for convenient chaining of method calls
      */
-    public RoguePanel write(String string, int x, int y, Color foreground, Color background) {
+      public RoguePanel write(String string, int x, int y, Color foreground, Color background) {
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RoguePanel.java]: The original spacing was correct here, I think.

File: src/edu/ucsb/cs56/projects/games/roguelike/RoguePanel.java [hunk 12]
```diff
@@ -579,7 +579,7 @@ public RoguePanel write(String string, int x, int y, Color foreground, Color bac
      * @param background the background color or null to use the default
      * @return this for convenient chaining of method calls
      */
-    public RoguePanel write(char character, int x, int y, Color foreground, Color background) {
+     public RoguePanel write(char character, int x, int y, Color foreground, Color background) {
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RoguePanel.java]: Ditto.

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java [hunk 13]
```diff
@@ -48,6 +48,24 @@
     //Matrix indicating which grid space has been discovered yet
     //1 = discovered, any other value = not discovered
     private int[][] discoveredArea;
+
+    // private levelChanged = logicHandler.getLevel(); //im not sure what im doing here
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: Kill your dead code.

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java (hunk 7 above)
Comment (mberlanga) [reply src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: This was for monster disappearing

File: src/edu/ucsb/cs56/projects/games/roguelike/GUI.java [hunk 14]
```diff
@@ -13,7 +13,7 @@
  */
 public class GUI
 {
-
+    private boolean gameRunning = false; //This boolean is used to make sure only one game instance is running, if true, menu will shut down and false will return menu
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/GUI.java]: Is this ever used?

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java (hunk 7 above)
Comment (xavierholt) [reply src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: This seems like a bandaid over a more fundamental problem...  What's the actual problem, and how does this fix it?

File: src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java [hunk 15]
```diff
@@ -363,26 +365,66 @@ public void keyPressed(KeyEvent key){
...
-
+	//Tell player they cannot run into a wall, else continue with game progress
+	if(logicHandler.getObject(x,y) instanceof Wall){
+	    canvas.write("     Ouch a Wall  ", 61,23,RoguePanel.yellow,RoguePanel.red);
 
-	checkPlayerStatus();
-	checkAllMonsterStatus();
-	trackDiscovery();
+	    //restores player (x,y) player position if a wall is hit
+	    while (!logicHandler.isGround(x, y)) {
```
Comment (xavierholt) [inline src/edu/ucsb/cs56/projects/games/roguelike/RogueController.java]: I don't really understand what this loop is doing...  Could it be an `else` for the previous `if` instead?

And if it's checking if a move is valid and moving you back, could you do this before the move happens in the first place?

**Pull request closed without merging**